ctapipe/tools/muon_reconstruction.py

```diff
@@ -5,121 +5,176 @@
+from astropy.coordinates import SkyCoord
 
 
-def _exclude_some_columns(subarray, writer):
-    """ a hack to exclude some columns of all output tables here we exclude
-    the prediction and mask quantities, since they are arrays and thus not
-    readable by pandas.  Also, prediction currently is a variable-length
-    quantity (need to change it to be fixed-length), so it cannot be written
-    to a fixed-length table.
-    """
-    all_camids = {str(x.camera) for x in subarray.tel.values()}
-    for cam in all_camids:
-        writer.exclude(cam, 'prediction')
-        writer.exclude(cam, 'mask')
-
-
-class MuonDisplayerTool(Tool):
+class MuonAnalysis(Tool):
     name = 'ctapipe-reconstruct-muons'
-    description = t.Unicode(__doc__)
+    description = Unicode(__doc__)
 
-    outfile = t.Unicode(
-        "muons.hdf5",
+    outfile = Unicode(
+        None,
+        allow_none=True,
         help='HDF5 output file name'
     ).tag(config=True)
 
-    display = t.Bool(
+    display = Bool(
         help='display the camera events', default=False
     ).tag(config=True)
 
-    classes = t.List([
-        CameraCalibrator, EventSource
-    ])
+    min_pixels = IntTelescopeParameter(
+        help=(
+            'Minimum number of pixels after cleaning and ring finding'
+            'required to process an event'
+        ),
+        default_value=10,
+    ).tag(config=True)
+
+    classes = [
+        CameraCalibrator, EventSource, MuonRingFitter, MuonIntensityFitter,
+    ]
 
-    aliases = t.Dict({
+    aliases = {
         'input': 'EventSource.input_url',
-        'outfile': 'MuonDisplayerTool.outfile',
-        'display': 'MuonDisplayerTool.display',
+        'outfile': 'MuonAnalysis.outfile',
         'max_events': 'EventSource.max_events',
         'allowed_tels': 'EventSource.allowed_tels',
-    })
+    }
 
     def setup(self):
-        self.source: EventSource = self.add_component(
-            EventSource.from_config(parent=self)
-        )
-        if self.source.input_url == '':
-            raise ToolConfigurationError("please specify --input <events file>")
+        self.source = self.add_component(EventSource.from_config(parent=self))
         self.calib = self.add_component(CameraCalibrator(
             parent=self,
             subarray=self.source.subarray
         ))
-        self.writer = self.add_component(HDF5TableWriter(self.outfile, "muons"))
+        self.ring_fitter = self.add_component(MuonRingFitter(
+            parent=self,
+        ))
+        self.intensity_fitter = self.add_component(MuonIntensityFitter(
+            parent=self,
+            subarray=self.source.subarray,
+        ))
+        self.cleaning = self.add_component(
+            TailcutsImageCleaner(
+                parent=self,
+                subarray=self.source.subarray,
+            )
+        )
+        if self.outfile:
+            self.writer = self.add_component(HDF5TableWriter(
+                self.outfile, "muons", add_prefix=True
```

Comment:
we could consider using the same hierarchy as the DL1 files and put the tables in the group `"dl1/event/telescope/parameters/muons"` instead, that might make merging files easier later (but pathnames inside a bit more complex!) That could come later though.

Reply:
How would the path for mc information look like? ANd is there somewhere I can read it up?

Reply:
There is discussion in #1066  and #1059 .  Right now the MC parameters are in `/dl1/event/subarray/mcshower` but since it's not really specific to DL1, we may move it elsewhere and probably rename to `true` instead of `mc`

Reply:
make an extra level for simulation values? `/sim/event/subarray/` `/sim/event/telescope/` ?

Reply:
yes that was one of the other suggestions